tools/AutomationTestFramework/RunAutomationTests.ps1

```diff
@@ -0,0 +1,128 @@
+# ----------------------------------------------------------------------------------
+    # Copyright Microsoft Corporation
+    # Licensed under the Apache License, Version 2.0 (the "License");
+    # you may not use this file except in compliance with the License.
+    # You may obtain a copy of the License at
+    # http://www.apache.org/licenses/LICENSE-2.0
+    # Unless required by applicable law or agreed to in writing, software
+    # distributed under the License is distributed on an "AS IS" BASIS,
+    # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    # See the License for the specific language governing permissions and
+    # limitations under the License.
+# ----------------------------------------------------------------------------------
+
+param (
+    [Parameter(Mandatory = $true)]
+    [string] $spPassword,
+    [bool] $createPackages = $true,
+    [bool] $uploadPackages = $true,
+    [bool] $processRunbooks = $true,
+    [bool] $waitForResults = $true
+)
+
+. "$PSScriptRoot\Management\PackageGenerator.ps1"
+. "$PSScriptRoot\Management\ModuleUploader.ps1"
+. "$PSScriptRoot\Management\RunbookProcessor.ps1"
+
+try {
+    $srcPath = "$PSScriptRoot\..\..\src"
+    $projectList = @('Compute', 'Network', 'Profile', 'Resources', 'Sql', 'Storage', 'Websites')
+    $testResourcesDir = "$PSScriptRoot\TestResources"
+    $packagingDir = "$PSScriptRoot\Package"
+    $helperModuleName = 'Smoke.Helper'
+    $testModuleName = 'Smoke.Tests'
+    $runbooksPath = "$PSScriptRoot\Runbooks"
+    $success = $false
+
+    $automation = @{
+        ResourceGroupName = 'azposjhautomation';
+        AccountName = 'azposhautomation'
+    }
+
+    $storage = @{
+        ResourceGroupName = 'transit2automation';
+        AccountName = 'transit2automation';
+        ContainerName = 'testsmodule'
+    }
+
+    $template = @{
+        SubscriptionName = 'Azure SDK Powershell Test';
+        AutomationConnectionName = 'AzureRunAsConnection';
+        Path = "$testResourcesDir\RunbookTemplate.ps1"
+    }
+
+    $signedModuleList = @('AzureRM.Resources', 'AzureRM.Compute', 'AzureRM.Automation', 'AzureRM.Network', 'AzureRM.Storage', 'AzureRM.Websites', 'AzureRM.KeyVault', 'AzureRM.Sql')
```

Comment:
I am hoping we can derive this list from the list of tested RPs above (for example, by gathering RequiredModules from the psd1 files).

Reply:
[As mentioned above](https://github.com/Azure/azure-powershell/pull/5301#discussion_r163945304), this was just part of the 'bubbling up' process. Signed modules can either be determined automatically be projects being ran, or by some kind of Include filter. We can discuss this further if you'd like.

Reply:
These two changes need to be  in concert, as mentioned above.  Since you are the team expert on this code, I'd like you to file an issue for the additional refactoring required to make this flexible enough to pass in an RP list, or to run *all* tests properly annotated